crypto/src/e2e_identity/rotate.rs

```diff
@@ -1067,5 +1067,89 @@ pub(crate) mod tests {
+                                .unwrap()
+                                .certificate;
+
+                            // Alice creates a new Credential, updating her handle/display_name
+                            let alice_cid = alice_central.mls_central.get_client_id();
+                            let (new_handle, new_display_name) = ("new_alice_wire", "New Alice Smith");
+                            alice_central
+                                .mls_central
+                                .rotate_credential(&case, new_handle, new_display_name, alice_og_cert, intermediate_ca)
+                                .await;
+
+                            // Verify old identity is a basic identity in the MLS group
+                            let alice_old_identities = alice_central
+                                .mls_central
+                                .get_device_identities(&id, &[alice_cid])
+                                .await
+                                .unwrap();
+                            let alice_old_identity = alice_old_identities.first().unwrap();
+                            assert_eq!(alice_old_identity.credential_type, MlsCredentialType::Basic);
+                            assert_eq!(alice_old_identity.x509_identity, None);
+
+                            // Alice issues an Update commit to replace her current identity
+                            let commit = alice_central.mls_central.e2ei_rotate(&id, None).await.unwrap();
+
+                            // Bob decrypts the commit...
+                            let decrypted = bob_central
+                                .mls_central
+                                .decrypt_message(&id, commit.commit.to_bytes().unwrap())
+                                .await
+                                .unwrap();
+                            // ...and verifies that now Alice is represented with her new identity
+                            alice_central.mls_central.verify_sender_identity(&case, &decrypted);
+
+                            // Finally, Alice merges her commit and verifies her new identity gets applied
+                            alice_central.mls_central.commit_accepted(&id).await.unwrap();
+                            alice_central
+                                .mls_central
+                                .verify_local_credential_rotated(&id, new_handle, new_display_name)
+                                .await;
+
+                            let final_count = alice_central.mls_central.count_entities().await;
+                            assert_eq!(init_count.encryption_keypair, final_count.encryption_keypair);
+                            assert_eq!(
+                                init_count.epoch_encryption_keypair,
+                                final_count.epoch_encryption_keypair
+                            );
+                            assert_eq!(init_count.key_package, final_count.key_package);
```

Comment:
This means a search for a basic credential key package would fail now, right?

Reply:
I mean this count doesn't distinguish. In this test the key packages aren't touched. 10 key packages are generated when the centrals are created but that's it.

I honestly now that I'm looking at it I don't see the point of asserting that the `key_package` or `epoch_encryption_keypair` count hasn't changed.

Reply:
Yeah, then I would omit those checks in order to avoid confusion. :+1: